pkg/types/validation/installconfig.go

```diff
@@ -395,6 +395,9 @@ func validateCompute(platform *types.Platform, control *types.MachinePool, pools
 			allErrs = append(allErrs, field.Invalid(poolFldPath.Child("architecture"), p.Architecture, "heteregeneous multi-arch is not supported; compute pool architecture must match control plane"))
 		}
 		allErrs = append(allErrs, ValidateMachinePool(platform, &p, poolFldPath)...)
+		if len(p.Workloads) > 0 {
```

Comment:
Can we do this check in `ValidateMachinePool` instead so that we don't report potential errors for the specific workloads in addition to reporting that workloads as a whole are not allowed? You could either pass in the type of machine pool being validated or rely on the name of the machine pool to distinguish.

Reply:
Good idea!  Moved.